### DISCUSSION TITLE
【战斗流程协议】如何根据城防炮进度条使用技能

### DISCUSSION THREAD

**Discussion by lucienshawls** [MaaAssistantArknights/MaaAssistantArknights#10373] [Q&A] [answered]
目前，由于我的box不足，我打燃烧街区需要借logos的助战，导致该剿灭无法代理，只能手动操作，我准备自己写一套战斗流程交给MAA自动打。

我使用了幽灵鲨来抗城防炮，即在城防炮进度条快满时开启2技能免死。但是MAA的战斗协议里似乎并没有基于城防炮进度条的触发条件。请问我应该如何配合使用其它触发条件来达成这一目的？

**Comment by Constrat** [comment, marked as answer]
Unfortunately, all the Artillery Defense tasks usually use Mudrock as she's completely automated for that exact reason. MAA has no triggers for the artillery cannon.
The cannon doesn't even shoot at specific times because enemies can charge it faster or slower depending on how many there are. 
But you could try to time it with the enemy kills and the post delay option to get the skill to active at the right time.
It's not easy but it should be doable:
![image](https://github.com/user-attachments/assets/f645bc85-c97e-4793-8e68-738c36738e80)

**Reply by Constrat** [reply]
cc @Saratoga-Official you're the expert on copilot homeworks, if you have other ideas :)

**Reply by lucienshawls** [reply]
> Unfortunately, all the Artillery Defense tasks usually use Mudrock as she's completely automated for that exact reason. MAA has no triggers for the artillery cannon. The cannon doesn't even shoot at specific times because enemies can charge it faster or slower depending on how many there are. But you could try to time it with the enemy kills and the post delay option to get the skill to active at the right time. It's not easy but it should be doable: ![image](https://private-user-images.githubusercontent.com/56174894/361766355-f645bc85-c97e-4793-8e68-738c36738e80.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ5MTk3NjMsIm5iZiI6MTcyNDkxOTQ2MywicGF0aCI6Ii81NjE3NDg5NC8zNjE3NjYzNTUtZjY0NWJjODUtYzk3ZS00NzkzLThlNjgtNzM4YzM2NzM4ZTgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI5VDA4MTc0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM0Y2YyZmE5M2Y3NzlhMzMzYWJkOTRmN2JmMTI4MzVjZmMxOTY2ZWU4MmQwMzNiNGU1ZmJlMzkyYzZmZThlYjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.sEip77DuUDIsX4ClkwnWrGceNrsBV-vs8vN7YjNS7KM)

Thank you for your answer. I managed to activate the skills at specific enemy kills with specific time delay. However this plan works for my account only because the time delay should vary based on the levels of the characters (the enemies may be eliminated ahead of time if the levels of the characters are too high). This could be acceptable, since the script was developed for my own personal use in the first place.

P.S. Right after the script passed all the tests and can be used stably, I sadly found out that the commision is soon closed and I will not need the script anymore. So why would I spend so much time on this script?

**Comment by Saratoga-Official** [comment]


**Reply by lucienshawls** [reply]
为什么不直接手打呢？因为手打需要手动开技能，每次打剿灭要花费十几分钟，每周要打五次，而MAA或许可以帮我节省这些精力，所以我希望让MAA自动帮我打，就是这么简单。

事实上我后来根据自己的干员排轴，手打第一遍掐秒表，然后写好脚本交MAA打一遍排错，再最后让MAA打一遍确认，事实上也没多花太多时间。

只不过，我不巧忘记关注这个剿灭的结束时间了，写完以后才发现这个剿灭委托马上结束了而已。

另外，希望您在现实中和人交流时一定要坚持使用您回复我的时候使用的语气，要不然无法体现您的聪明才智。

**Reply by lucienshawls** [reply]
恕我没有大杯理解，我的阵容需要借用好友logos才能打过。这一点我在这篇讨论一开始就说明了。

P.S. 以防您不知道，借了助战的话是不能代理的。

**Reply by lucienshawls** [reply]
对于代理问题，你给出的解决方法是使用低配，这当然是合理的办法；而我选择借用助战，那么面对无法代理的问题，我选择了使用MAA自动处理，并尝试自己编写脚本，并未做伸手党。在编写脚本的过程中，我遇到了困难，并在这里发了讨论，之后明白了我必须使用一些其它方法来写脚本。

在我已经明白以上情况、并将拨冗为我解答的大佬的回复标记为解答之后，您在没有完整阅读本讨论的情况下，直接向我提出了质疑，认为我这么做没有意义。且不论您没有认真阅读讨论内容这一点，您提出的“使用低配”对该讨论没有实质上的帮助，既不能告知我MAA识别城防炮的方式，也不能告诉我如何用其它方法变相达到根据城防炮进度而作出动作的方法，我认为有离题之嫌。

对于语气问题，即使您可能是无意的，但是我觉得，您使用的措辞是强硬的、无可置疑的，这样的语气应当可以被理解为不希望心平气和地讨论。我为我的这一想法向您道歉。

**Reply by Saratoga-Official** [reply]
> 对于代理问题，你给出的解决方法是使用低配，这当然是合理的办法；而我选择借用助战，那么面对无法代理的问题，我选择了使用MAA自动处理，并尝试自己编写脚本，并未做伸手党。在编写脚本的过程中，我遇到了困难，并在这里发了讨论，之后明白了我必须使用一些其它方法来写脚本。
> 
> 在我已经明白以上情况、并将拨冗为我解答的大佬的回复标记为解答之后，您在没有完整阅读本讨论的情况下，直接向我提出了质疑，认为我这么做没有意义。且不论您没有认真阅读讨论内容这一点，您提出的“使用低配”对该讨论没有实质上的帮助，既不能告知我MAA识别城防炮的方式，也不能告诉我如何用其它方法变相达到根据城防炮进度而作出动作的方法，我认为有离题之嫌。
> 
> 对于语气问题，即使您可能是无意的，但是我觉得，您使用的措辞是强硬的、无可置疑的，这样的语气应当可以被理解为不希望心平气和地讨论。我为我的这一想法向您道歉。

关于解决方法，已经有人提供了我没有必要重复说，我看到这个讨论是给你提供解决方法的佬@我让我补充而已，我就没看之前的讨论，这确实是我的错，我的语气也是平时解决问题解决的多造成的，你不能指望一天解决好几个问题的人能心平气和的说话，但你阴阳怪气我确实看着很不爽
